main.go

```diff
@@ -148,6 +182,27 @@ func main() {
 			return
 		}
 
+		//If proxy is set to kubernetes service(apiserver) add SA token to the request
+		if strings.Contains(cfg.upstream, "kubernetes.default") {
+			director := proxy.Director
+			proxy.Director = func(req *http.Request) {
+				director(req)
+				req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", kcfg.BearerToken))
+			}
+		}
+
+		if cfg.upstreamInsecure {
```

Comment:
This change seems to be unrelated to the actual OIDC contribution as it adds the possibility to skip TLS verification for upstream. Two comments on this:

1. This `if` statement configures the proxy for every request which seems unnecessary and potentially racy. Do you mind to factor this somewhere around line 250?

2. The proxy transport gets also overwritten if h2c is enabled, so `upstreamInsecure` and `upstreamForceH2C` are mutually exclusive. We have to take care to initialize the http transport if `upstreamInsecure=true` _before_ the h2c 
check in line 250.

Reply:
Yes, you're right, skipping tls for the upstream will be removed